src/middlewared/middlewared/utils/pam.py

```diff
@@ -0,0 +1,302 @@
+    # Subset of possible actions for control keys in pam line
+    IGNORE = 'ignore'
+    BAD = 'bad'
+    DIE = 'die'
+    OK = 'ok'
+    DONE = 'done'
+    RESET = 'reset'
+
+
+@dataclass(slots=True, frozen=True)
+class PAMControl:
+    response: PAMResponse
+    action: PAMAction | int
+
+    def as_conf(self):
+        return f'{self.response}={self.action}'
+
+
+@dataclass(slots=True, frozen=True)
+class PAMLine:
+    pam_service: PAMService
+    pam_control: PAMSimpleControl | tuple[PAMControl]
+    pam_module: PAMModule
+    pam_module_args: tuple[str] | None = None
+
+    def __dump_control(self):
+        if isinstance(self.pam_control, PAMSimpleControl):
+            return self.pam_control
+
+        return f'[{" ".join(ctrl.as_conf() for ctrl in self.pam_control)}]'
+
+    def as_conf(self):
+        if self.pam_module_args is None:
+            return '\t'.join([
+                self.pam_service,
+                self.__dump_control(),
+                self.pam_module
+            ])
+
+        return '\t'.join([
+            self.pam_service,
+            self.__dump_control(),
+            self.pam_module,
+            ' '.join(self.pam_module_args)
+        ])
+
+
+@dataclass(slots=True, frozen=True)
+class PAMConfLines:
+    service: PAMService
+    primary: tuple[PAMLine]
+    secondary: tuple[PAMLine] | None = None
+
+
+# Below this point are intialized PAM configuration objects for use in mako files
+
+STANDALONE_ACCOUNT = PAMConfLines(
+    service=PAMService.ACCOUNT,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 1),
+                PAMControl(PAMResponse.NEW_AUTHTOK_REQD, PAMAction.DONE),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+    )
+)
+
+AD_ACCOUNT = PAMConfLines(
+    service=PAMService.ACCOUNT,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
+                PAMControl(PAMResponse.NEW_AUTHTOK_REQD, PAMAction.DONE),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 1),
+                PAMControl(PAMResponse.NEW_AUTHTOK_REQD, PAMAction.DONE),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.WINBIND,
+            pam_module_args=('krb5_auth', 'krb5_ccache_type=FILE')
+        )
+    )
+)
+
+SSS_ACCOUNT = PAMConfLines(
+    service=PAMService.ACCOUNT,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
```

Comment:
In current code, attached to the FreeIPA I see the following in `common-account`:
`account [success=1 new_authtok_reqd=done default=ignore]        pam_unix.so`
Maybe this line should have `1` and not `2`?